nixos/modules/system/boot/loader/limine/limine-install.py

```diff
@@ -0,0 +1,378 @@
+    dest_file = f'{package_id}-{suffix}'
+    dest_path = os.path.join(limine_dir, target, dest_file)
+
+    if not os.path.exists(dest_path):
+      copy_file(path, dest_path)
+
+    path_with_prefix = os.path.join('/limine', target, dest_file)
+    result = f'boot()://{path_with_prefix}'
+
+  if config('validateChecksums'):
+    with open(path, 'rb') as file:
+      b2sum = hashlib.blake2b()
+      b2sum.update(file.read())
+
+      result += f'#{b2sum.hexdigest()}'
+
+  return result
+
+
+def get_path_uri(path: str) -> str:
+  return get_copied_path_uri(path, "")
+
+
+def get_file_uri(profile: str, gen: Optional[str], spec: Optional[str], name: str) -> str:
+  gen_path = get_system_path(profile, gen, spec)
+  path_in_store = os.path.realpath(os.path.join(gen_path, name))
+  return get_path_uri(path_in_store)
+
+
+def get_kernel_uri(kernel_path: str) -> str:
+  return get_copied_path_uri(kernel_path, "kernels")
+
+
+def config_entry(levels: int, toplevel: str, label: str, comment: str, kernel_path: str, kernel_params: str, init: str, initrd_path: Optional[str], initrd_secrets: Optional[str]) -> str:
+  entry = '/' * levels + label + '\n'
+  entry += f'comment: {comment}\n'
+  entry += 'protocol: linux\n'
+  entry += 'kernel_path: ' + get_kernel_uri(kernel_path) + '\n'
+  entry += ('cmdline: ' + 'init=' + init + ' ' + kernel_params).strip() + '\n'
+  if initrd_path:
+    entry += f'module_path: ' + get_kernel_uri(initrd_path) + '\n'
+
+  if initrd_secrets:
+    initrd_secrets_path = limine_dir + '/kernels/' + os.path.basename(toplevel) + '-secrets'
+    os.makedirs(initrd_secrets_path)
+
+    old_umask = os.umask()
+    os.umask(0o137)
+    initrd_secrets_path_temp = tempfile.mktemp(os.path.basename(toplevel) + '-secrets')
+
+    if os.system(initrd_secrets + " " + initrd_secrets_path_temp) != 0:
+        print(f'warning: failed to create initrd secrets for "{label}"', file=sys.stderr)
+        print(f'note: if this is an older generation there is nothing to worry about')
+
+    if os.path.exists(initrd_secrets_path_temp):
+        copy_file(initrd_secrets_path_temp, initrd_secrets_path)
+        os.unlink(initrd_secrets_path_temp)
+        entry += 'module_path: ' + get_kernel_uri(initrd_secrets_path) + '\n'
+
+    os.umask(old_umask)
+  return entry
+
+
+def generate_config_entry(profile: str, gen: str) -> str:
+  boot_spec = json.load(open(get_system_path(profile, gen) + '/boot.json', 'r'))
+
+  kernel_path = boot_spec['org.nixos.bootspec.v1']['kernel']
+  initrd_path = boot_spec['org.nixos.bootspec.v1'].get('initrd')
+  initrd_secrets = boot_spec['org.nixos.bootspec.v1'].get('initrdSecrets')
+  init_path = boot_spec['org.nixos.bootspec.v1']['init']
+  kernel_params = ' '.join(boot_spec['org.nixos.bootspec.v1']['kernelParams'])
+
+  entry = config_entry(2, boot_spec['org.nixos.bootspec.v1']['toplevel'], f'Generation {gen}', boot_spec['org.nixos.bootspec.v1']['label'], kernel_path, kernel_params, init_path, initrd_path, initrd_secrets)
```

Comment:
Both the current Grub and Systemd-boot scripts include the build date of the generation to the bootloader entries. Both of them get the information from `stat`ing the profile link, though the information is also available from `nix-env --list-generations`, and from the `registrationTime` field of `nix path-info --json`.

Limine could put it in either the entry label or the `comment`. It might also be worth keeping #366958 in mind, using a method that's compatible with `date` format specifiers.